MapboxCoreNavigationTests/NavigationServiceTests.swift

```diff
@@ -122,54 +122,78 @@ class NavigationServiceTests: XCTestCase {
         let firstLocation = dependencies.routeLocations.firstLocation
 
         navigation.locationManager!(navigation.locationManager, didUpdateLocations: [firstLocation])
-        XCTAssertEqual(navigation.router.location!.coordinate, firstLocation.coordinate, "Check snapped location is working")
+        
+        // Check snapped location is working
+        XCTAssertEqual(navigation.router.location!.coordinate.latitude, firstLocation.coordinate.latitude, accuracy: 0.005, "Latitudes should be almost equal")
+        XCTAssertEqual(navigation.router.location!.coordinate.longitude, firstLocation.coordinate.longitude, accuracy: 0.005, "Longitudes should be almost equal")
 
         let firstCoordinateOnUpcomingStep = navigation.router.routeProgress.currentLegProgress.upcomingStep!.shape!.coordinates.first!
         let firstLocationOnNextStepWithNoSpeed = CLLocation(coordinate: firstCoordinateOnUpcomingStep, altitude: 0, horizontalAccuracy: 10, verticalAccuracy: 10, course: 10, speed: 0, timestamp: Date())
 
         navigation.locationManager!(navigation.locationManager, didUpdateLocations: [firstLocationOnNextStepWithNoSpeed])
-        XCTAssertEqual(navigation.router.location!.coordinate, navigation.router.routeProgress.currentLegProgress.currentStep.shape!.coordinates.last!, "When user is not moving, snap to current leg only")
+        
+        let lastCoordinate = navigation.router.routeProgress.currentLegProgress.currentStep.shape!.coordinates.last!
+        
+        // When user is not moving, snap to current leg only
+        XCTAssertEqual(navigation.router.location!.coordinate.latitude, lastCoordinate.latitude, accuracy: 0.005, "Latitudes should be almost equal")
+        XCTAssertEqual(navigation.router.location!.coordinate.longitude, lastCoordinate.longitude, accuracy: 0.005, "Longitudes should be almost equal")
 
         let firstLocationOnNextStepWithSpeed = CLLocation(coordinate: firstCoordinateOnUpcomingStep, altitude: 0, horizontalAccuracy: 10, verticalAccuracy: 10, course: 10, speed: 5, timestamp: Date())
         navigation.locationManager!(navigation.locationManager, didUpdateLocations: [firstLocationOnNextStepWithSpeed])
 
-        XCTAssertEqual(navigation.router.location!.coordinate.latitude, firstCoordinateOnUpcomingStep.latitude, accuracy: 0.0005, "User is snapped to upcoming step when moving")
-        XCTAssertEqual(navigation.router.location!.coordinate.longitude, firstCoordinateOnUpcomingStep.longitude, accuracy: 0.0005, "User is snapped to upcoming step when moving")
+        // User is snapped to upcoming step when moving
+        XCTAssertEqual(navigation.router.location!.coordinate.latitude, firstCoordinateOnUpcomingStep.latitude, accuracy: 0.005, "Latitudes should be almost equal")
```

Comment:
The other question that comes from the question above: should we make accuracy threshold lower?

Reply:
A difference of 0.005° longitude is more than 100&nbsp;meters off at the latitudes in this test fixture, whereas 0.0005° is within a dozen meters or so. So I think the accuracy threshold does need to be lower, maybe nearly as low as it was previously.

Reply:
In previous versions of nav. native right after starting navigation `initialized` `routeState` was returned by `RouteController`. This meant that `rawLocation` was used, which is last location, as a result no accuracy checks were needed at that time. Example with `routeState` values with 3 locations:
```
1. Current route state: initialized
2. Current route state: initialized
3. Current route state: tracking
```
As per discussion with @SiarheiFedartsou in nav. native 14.1.0 active guidance logic was changed and now `initialized` state is only returned in `setRouteForRouteResponse(_:route:leg:)` (I've checked to make sure and that's correct). This means that now route state has such sequence:
```
1. Current route state: tracking
2. Current route state: tracking
3. Current route state: tracking
```
This means that instead of `rawLocation` we'll be using `snappedLocation` because of such logic in `RouteController`:
```swift
var snappedLocation: CLLocation? {
    let status = navigator.getStatusForTimestamp(Date())
    guard status.routeState == .tracking || status.routeState == .complete else {
        return nil
    }
    
    return CLLocation(status.location)
}
...
public var location: CLLocation? {
    return snappedLocation ?? rawLocation
}
```

Since `snappedLocation` is returned from `MBNavigator` it explains the fact that last location and `snappedLocation` are not the same (that is the reason accuracy was introduced). I can see two solutions for now:
1. Leaving accuracy argument, but slightly change it to contain more 'realistic' value, e.g. distance between `snappedLocation` and last location?
2. Changing logic in `RouteController` to apply new flow in nav. native. E.g. somehow detecting that we're in `initialized` state even though we're currently already in `tracking` as per status returned from nav. native. 

I might still miss some pitfalls, so if you have any ideas please let me know.

Reply:
Thanks for the comprehensive analysis!

Seems the first option is quicker, but it looks more like cutting the corner.
Should we spend time on trying to update the `RouteController`?

Reply:
I've updated tests to use original accuracy of `0.0005°` for coordinates and added distance check between coordinates (set to default accuracy of `1` meter). Had to update several tests to reflect current behavior of logic in nav. native `14.1.2`. Please check updated comments in changed tests.